### PR TITLE
urfave v2 shutdown patterns

### LINKED ISSUE #6742: op-service: opio change to use cleaner urfave v2 shutdown pattern
- [ ] `opio` waits for interrupt signal, but doesn’t cleanly close urfave CLI, we should use `app.RunContext` (urfave v2) and close the context on interrupt.
- [ ] Update the main entrypoints of all services to use the cleaner shutdown pattern.
- [ ] Force-exit on e.g. receiving 10 consecutive interrupts.

### PR DESCRIPTION
**Pull request by ethenotethan**
**Description**
TBD - Still WIP

A clear and concise description of the features you're adding in this pull request.

**Tests**

Please describe any tests you've added. If you've added no tests, or left important behavior untested, please explain why not.

**Additional context**
TBD - Still WIP

**Metadata**

- Fixes #6742

### CODE REVIEW

File: op-node/cmd/main.go [hunk 1]
```diff
@@ -147,7 +165,10 @@ func RollupNodeMain(ctx *cli.Context) error {
 				log.Error("heartbeat goroutine crashed", "err", err)
 			}
 		}()
-		defer beatCtxCancel()
+		close = func() {
+			close()
```
Comment (protolambda) [inline op-node/cmd/main.go:169]: this seems like it would indefinitely recurse?

File: op-node/cmd/main.go [hunk 2]
```diff
@@ -158,11 +179,12 @@ func RollupNodeMain(ctx *cli.Context) error {
 				log.Error("error starting pprof", "err", err)
 			}
 		}()
-		defer pprofCancel()
+		close = func() {
+			close()
```
Comment (protolambda) [inline op-node/cmd/main.go:183]: same issue here

File: op-batcher/cmd/main.go [hunk 3]
```diff
@@ -28,24 +32,33 @@ func main() {
...
 	if err != nil {
 		log.Crit("Application failed", "message", err)
 	}
 }
 
 // curryMain transforms the batcher.Main function into an app.Action
-// This is done to capture the Version of the batcher.
-func curryMain(version string) func(ctx *cli.Context) error {
+// This is done to capture the Version and closure of the batcher.
+func curryMain(cancel func(), version string) func(ctx *cli.Context) error {
```
Comment (protolambda) [inline op-batcher/cmd/main.go:51]: I think it's better if we de-duplicate this code

File: indexer/cmd/indexer/main.go [hunk 4]
```diff
@@ -12,8 +14,17 @@ var (
 )
 
 func main() {
+	ctx, cancel := context.WithCancel(context.Background())
+
+	// Spinup a goroutine to catch interrupts and cancel the application context.
+	go func() {
+		opio.BlockOnInterrupts()
+		log.Crit("caught interrupt, shutting down...")
```
Comment (trianglesphere) [inline indexer/cmd/indexer/main.go:22]: log.Crit will panic. We don't use it except in rare code where we don't do error handling (and this code should do proper error handling).

**Pull request closed without merging**